src/sb_pointcloud_processing/src/IGVCVisualizerNode.cpp

```diff
@@ -0,0 +1,152 @@
+        pcl::fromPCLPointCloud2(*pcl_input, *pcl_rgb);
+
+        // Remove NaNs from point cloud
+        std::vector<int> indices;
+        pcl::removeNaNFromPointCloud(*pcl_rgb, *pcl_rgb, indices);
+
+        // Renew pcl point data
+        visualized_cloud = pcl_rgb;
+
+        // Remove all old point clouds from the view
+        viewer->removeAllPointClouds(0);
+
+        // Show the new point cloud retrieved from the camera
+        viewer->addPointCloud(visualized_cloud, "visualized_cloud");
+
+        // Setup how big the points are in the point cloud
+        viewer->setPointCloudRenderingProperties(pcl::visualization::PCL_VISUALIZER_POINT_SIZE, 1.0,
+                                                 "visualized_cloud");
+    }
+}
+
+void IGVCVisualizerNode::updateVisualizerCallback(const ros::TimerEvent &event) {
+    // Update the viewer
+    viewer->spinOnce(100);
+    boost::this_thread::sleep(boost::posix_time::microseconds(100000));
+}
+
+boost::shared_ptr<pcl::visualization::PCLVisualizer> IGVCVisualizerNode::setUpPCLVisualizer() {
+    boost::shared_ptr<pcl::visualization::PCLVisualizer> viewer(new pcl::visualization::PCLVisualizer("PCL Viewer"));
+    viewer->registerPointPickingCallback(pointPickEventOccurred, (void *) viewer.get());
+    viewer->registerKeyboardCallback(keyboardEventOccurred, (void *) viewer.get());
+    viewer->initCameraParameters();
+
+    return viewer;
+}
+
+void IGVCVisualizerNode::retrieveVisualizerParameters(ros::NodeHandle private_nh) {
+    float default_h_margin_of_error = 10;
+    SB_getParam(private_nh, "h_margin_of_error", h_margin_of_error, default_h_margin_of_error);
+
+    float default_s_margin_of_error = 0.1;
+    SB_getParam(private_nh, "s_margin_of_error", s_margin_of_error, default_s_margin_of_error);
+
+    float default_v_margin_of_error = 0.1;
+    SB_getParam(private_nh, "v_margin_of_error", v_margin_of_error, default_v_margin_of_error);
+
+    float default_image_update_rate = 0.1;
+    SB_getParam(private_nh, "image_update_rate", image_update_rate, default_image_update_rate);
+}
+
+void IGVCVisualizerNode::updateFilterParams(float h, float s, float v) {
+    setParameter("hue_filter", "filter_limit_min", h - h_margin_of_error);
+    setParameter("hue_filter", "filter_limit_max", h + h_margin_of_error);
+    setParameter("saturation_filter", "filter_limit_min", s - s_margin_of_error);
+    setParameter("saturation_filter", "filter_limit_max", s + s_margin_of_error);
+    setParameter("value_filter", "filter_limit_min", v - v_margin_of_error);
+    setParameter("value_filter", "filter_limit_max", v + v_margin_of_error);
+}
+
+void IGVCVisualizerNode::setParameter(std::string node_name, std::string param_name, float val) {
+    // Run command line arguments to dynamically reconfigure parameter
+    std::string set_command = std::string("rosrun dynamic_reconfigure dynparam set");
+    system((set_command + " " + node_name + " " + param_name + " " + std::to_string(val)).c_str());
```

Comment:
Is there no way to do this programmatically? (ie. calling dynamic reconfigure directly instead of running a command from cmd line)?

Reply:
Dynamic_reconfigure only has a library for python, and not C++.  I just followed the instructions at the bottom of this page http://wiki.ros.org/hokuyo_node/Tutorials/UsingDynparamToChangeHokuyoLaserParameters#PythonAPI